lib/financials.js

```diff
@@ -104,17 +104,30 @@ async function getFinancialDataFromServer(): Promise<FinancialDataShapeType> {
     AsyncStorage.setItem(FINANCIALS__CACHE__BALANCES, JSON.stringify(new Date())),
   ])
 
-  return {flex, ole, print}
+  return {flex, ole, print};
 }
 
 
 // TODO: come up with a better story around auth for olecard
-export default async function getWeeklyMealsRemaining() {
+export async function getWeeklyMealsRemaining() {
   let {username, password} = await loadLoginCredentials()
   let form = buildFormData({
     username: username,
     password: password,
   })
-  let loginResult = await api.post('/apps/olecard/checkbalance/authenticate.cfm', {body: form})
-  return loginResult
+  let fetchResult = await api.post('/apps/olecard/checkbalance/authenticate.cfm', {body: form})
+
+  try {
+    let dom = parseHtml(fetchResult.body)
+    let data = cssSelect('.accountrow', dom)
+    let values = []
+    data.map((item) => {
+      values.push(item.next.next.children[0].data)
+    })
+    return({"weeklyMeals": values[4], "dailyMeals": values[3]})
+  } catch (error) {
+    console.log("error in fetching meal data")
```

Comment:
Maybe we should present an alert, or some information for the user to know that it could not be fetched?